src/plotting/tests/test_plotter.py

```diff
@@ -28,37 +28,55 @@ def test__create_plot(self):
         self.assertEqual(plotter._insert_data_in_axs.call_count, 1)
         self.assertEqual(plotter._save_plot.call_count, 1)
 
-    def test__insert_data_in_axs(self):
+    @patch('src.plotting.plotter.DataFileHandler.add_dataframes_from_folders_as_mean')
+    @patch('src.plotting.plotter.DataHandler.prepare_dataframe_for_boxplot')
+    def test__insert_data_in_axs(self, mock_data_file_handler, mock_data_handler):
         plotter = Plotter({})
-        plotter._create_subplot_df = MagicMock(return_value=pd.DataFrame({"class": ["class1", "class2"], "value": [1, 2]}))
         plotter._add_subplot_df_to_axs = MagicMock()
-        plotter._create_subplot_paths_group = MagicMock()
-        plotter._insert_data_in_axs(["test", "test"], None)
+        plotter._insert_data_in_axs([["path1"], ["path2"]], [0], {"data": ["data1"]})
 
-        self.assertEqual(plotter._create_subplot_df.call_count, 2)
+        self.assertEqual(mock_data_file_handler.call_count, 2)
+        self.assertEqual(mock_data_handler.call_count, 2)
         self.assertEqual(plotter._add_subplot_df_to_axs.call_count, 2)
 
     def test__save_plot(self):
```

Comment:
Does this function save the plot to a local file? Is it feasible to actually save that file, and test whether it exists?

Reply:
The function itself yes, the test function no.
The function itself just does 3 calls from matplotlib:

- tight_layout
- savefig (which saves the plot to a local file)
- show (open the plot in a new window)
I don't think it is in the scope of this test to check if the file is created or not, since that should be already tested by matplotlib.

Reply:
Right, but instead of doing the 3 mock checks, wouldn't it be simpler to just run the function without mocking, and check whether the file has been generated? Or does "creating the file" not imply those 3 listed calls?

Reply:
The only call generating the file is the second one. And to check if the file is created doesn't prove that it worked as expected. In order to check that the function works as expected, we would need to:
- Add specific data to a dataframe
- Plot it
- Save the plot as a `resources` for the test
- Each time we run the test, we compare the generated file with the previously saved file

And again, I don't think it is our responsibility to test if the `savefig` method works at that level.

Reply:
Oh, no, absolutely, we are not supposed to test if `savefig` works. I was just think, perhaps the `savefig` method is called if (and only if) all the others work as expected. Then the assertion could be as easy as that. Got it.

Well, then just a future reference (feel free to use, or don't) which is this [`mock.assert_called_once_with`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once_with).

Reply:
Yes, sometime I use that, sometimes I use the call_count to 1. Depending on which one is closer to copy-paste heh.